wordpress-mobile/WordPress-iOS
# Issue/12697 fix default blog

Fixes #12697 

This is one possible approach to resolving #12697.  I had it mostly written before I started second guessing whether it would just be more straightforward to fetch account details a second time during auth. It seems a little wasteful, but this patch seems a little heavy handed. 🤷‍♂   Maybe there is another option?  Maybe syncing details is actually the way to go? Open to other ideas. 

This PR bumps the model revision to 91 and adds a new field to WPAccount: `primaryBlogID` which is set when account details are synced. ~When an account is saved, it attempts to set its defaultBlog relation if one doesn't already exist.  Its a small amount of overhead on every account save.~ When syncing account details, or when syncing blogs the default blog is updated, and the app extensions are configured.

To test: 
- Start with a fresh install. 
- Log into the app.
- As soon as you're logged in background the app and attempt to use the share extension via Safari. 
- Confirm you don't receive an error.

For a more granular test:
- Start with a fresh install. 
- Set a break point in `WPAccount setupAppExtensionsWithDefaultAccount`.
- Log in and confirm this method is called during the login flow.
- Visit the Me tab (which syncs account details), and confirm the method is called.
- Visit the blog list (which syncs blogs) and confirm this method is called.

cc @leandroalonso in case we decide to land this PR, cos you have another migration in the works. 

Update release notes:
